okhttp/src/main/java/okhttp3/internal/platform/Android10Platform.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal.platform;
+
+import android.annotation.SuppressLint;
+import android.net.ssl.SSLSockets;
+import android.os.Build;
+import java.util.List;
+import javax.annotation.Nullable;
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.SSLSocket;
+import okhttp3.Protocol;
+import org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement;
+
+/** Android 10+. */
+@SuppressLint("NewApi")
+class Android10Platform extends AndroidPlatform {
+  Android10Platform(Class<?> sslParametersClass) {
+    super(sslParametersClass, null, null, null, null, null);
+  }
+
+  @SuppressLint("NewApi")
+  @IgnoreJRERequirement
+  @Override public void configureTlsExtensions(
+      SSLSocket sslSocket, String hostname, List<Protocol> protocols) {
```

Comment:
Might be able to get rid of the hostname thing altogether in our master branch

Reply:
(no action to take on this PR)

Reply:
It can't go completely.

Compare

Android 10 it's gone completely

https://github.com/square/okhttp/blob/ff0dee1c34fc06084357c93dc480f6ad89fef24a/okhttp/src/main/java/okhttp3/internal/platform/android/AndroidSocketAdapter.kt#L60

Android it's needed for 21-24.

https://github.com/square/okhttp/blob/ff0dee1c34fc06084357c93dc480f6ad89fef24a/okhttp/src/main/java/okhttp3/internal/platform/android/Android10SocketAdapter.kt#L49

Reply:
I was overeager last time and it resulted in you releasing 4.3.1

https://github.com/square/okhttp/pull/5702/files